sci-mathematics/netgen/netgen-5.3.1.ebuild

```diff
@@ -0,0 +1,103 @@
+# Copyright 1999-2015 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+
+inherit autotools eutils flag-o-matic multilib versionator
+
+MY_PN=${PN}-mesher
+MY_PV=$(get_version_component_range 1-2)
+DESCRIPTION="NETGEN is an automatic 3d tetrahedral mesh generator"
+HOMEPAGE="http://www.hpfem.jku.at/netgen/"
+SRC_URI="mirror://sourceforge/project/${MY_PN}/${MY_PN}/${MY_PV}/${P}.tar.gz"
+
+SLOT="0"
+LICENSE="LGPL-2.1"
+KEYWORDS="~amd64 ~x86"
+IUSE="-ffmpeg jpeg -mpi opencascade"
+
+DEPEND="
+	dev-tcltk/tix
+	dev-tcltk/togl:1.7
+	virtual/opengl
+	x11-libs/libXmu
+	opencascade? ( sci-libs/opencascade:* )
+	ffmpeg? ( media-video/ffmpeg )
+	jpeg? ( virtual/jpeg:0= )
+	mpi? ( virtual/mpi ( || ( sci-libs/parmetis <sci-libs/metis-5.0 ) ) sci-libs/hdf5[mpi] ) "
+RDEPEND="${DEPEND}"
+# Note, MPI has not be tested.
+
+src_prepare() {
+	# Adapted from http://sourceforge.net/projects/netgen-mesher/forums/forum/905307/topic/5422824
+	epatch "${FILESDIR}/${PN}-5.x-missing-define.patch"
+	epatch "${FILESDIR}/${PN}-5.x-metis-numflag.patch"
+	epatch "${FILESDIR}/${PN}-5.x-occ-stl-api-change.patch"
```

Comment:
This is due to the way you called diff. In general, patches need to with respect to `${WORKDIR}`, not `${S}`.